src/NServiceBus.Persistence.CosmosDB.TransactionalSession/CosmosTransactionalSessionExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace NServiceBus.TransactionalSession
+{
+    using Configuration.AdvancedExtensibility;
+    using Features;
+
+    /// <summary>
+    /// Enables the transactional session feature.
+    /// </summary>
+    public static class CosmosTransactionalSessionExtensions
+    {
+        /// <summary>
+        /// Enables transactional session for this endpoint.
+        /// </summary>
+        public static PersistenceExtensions<CosmosPersistence> EnableTransactionalSession(
+            this PersistenceExtensions<CosmosPersistence> persistenceExtensions)
+        {
+            persistenceExtensions.GetSettings().EnableFeatureByDefault<CosmosTransactionalSession>();
```

Comment:
not that it matters too much, but is there are reason this is using `EnableFeatureByDefault` rather than `EnableFeature` given there is no valid use case to call both `EnableTransactionalSession()` AND `config.DisableFeature<CosmosTransactionalSession>()` ?

Reply:
Haven't thought about it from that angle. Good point. Will change to `EnableFeature`

Reply:
Ah EnableFeature is internal. So I would have to switch to `settings.Set(featureType.FullName, FeatureState.Enabled);`. Probably using the public API is the better choice here no?